rasa/core/featurizers/single_state_featurizer.py

```diff
@@ -0,0 +1,244 @@
+
+        # there is always either INTENT or ACTION_NAME
+        name_attribute = next(
+            (
+                attribute
+                for attribute in attributes
+                if attribute in {INTENT, ACTION_NAME}
+            ),
+            None,
+        )
+        if name_attribute and name_attribute not in output:
+            # nlu pipeline didn't create features for user or action
+            output[name_attribute] = self._create_features(
+                sub_state, name_attribute, sparse
+            )
+
+        return output
+
+    def encode_state(
+        self, state: State, interpreter: Optional[NaturalLanguageInterpreter]
+    ) -> Dict[Text, List["Features"]]:
+        state_features = {}
+        for state_type, sub_state in state.items():
+            if state_type == PREVIOUS_ACTION:
+                state_features.update(
+                    self._extract_state_features(sub_state, interpreter, sparse=True)
+                )
+            # featurize user only if it is "real" user input,
+            # i.e. input from a turn after action_listen
+            if state_type == USER and prev_action_listen_in_state(state):
+                state_features.update(
+                    self._extract_state_features(sub_state, interpreter, sparse=True)
+                )
+                if sub_state.get(ENTITIES):
+                    state_features[ENTITIES] = self._create_features(
+                        sub_state, ENTITIES, sparse=True
+                    )
+
+            if state_type in {SLOTS, ACTIVE_LOOP}:
+                state_features[state_type] = self._create_features(
+                    sub_state, state_type, sparse=True
+                )
+
+        return state_features
+
+    def _encode_action(
+        self, action: Text, interpreter: Optional[NaturalLanguageInterpreter]
+    ) -> Dict[Text, List["Features"]]:
+
+        if action in self.action_texts:
+            action_as_sub_state = {ACTION_TEXT: action}
+        else:
+            action_as_sub_state = {ACTION_NAME: action}
+
+        return self._extract_state_features(action_as_sub_state, interpreter)
+
+    def encode_all_actions(
+        self, domain: Domain, interpreter: Optional[NaturalLanguageInterpreter]
+    ) -> List[Dict[Text, List["Features"]]]:
+
+        return [
+            self._encode_action(action, interpreter) for action in domain.action_names
+        ]
+
+
+class BinarySingleStateFeaturizer(SingleStateFeaturizer):
+    def __init__(self) -> None:
+        super().__init__()
+        common_utils.raise_warning(
+            f"'{self.__class__.__name__}' is deprecated and "
+            f"will be removed in the future. "
+            f"It is recommended to use the '{SingleStateFeaturizer.__name__}' instead.",
+            category=DeprecationWarning,
+            docs=DOCS_URL_MIGRATION_GUIDE,
+        )
+
+    def encode_state(
+        self, state: State, interpreter: Optional[NaturalLanguageInterpreter]
```

Comment:
but interpreter (not None) will be passed to it. We want to make sure that it is not used to mimic old behavior

Reply:
> We want to make sure that it is not used to mimic old behavior

That makes total sense. 

I'd challenge how the classes are structured: I think there should be a subclass of `SingleStateFeaturizer` which explicitly checks that an interpreter was given and raise an error in case it wasn't. Supporting evidence for me is that `if interpreter is not None` check in line 118. It shows that `SingleStateFeaturizer` is handling logic for both, featurizers who need an `interpreter` and featurizers who don't have `interpreter`s. The subclass can implement a `get_output` which can expect the interpreter to be available and return the stuff from lines 119-141.

Reply:
> It shows that SingleStateFeaturizer is handling logic for both, featurizers who need an interpreter and featurizers who don't have interpreters.

this is intended

Actually, I think it can never be `None` except when we set it to `None`. I'll try to refactor it

Reply:
basically, going in the future, we wouldn't have a featurizer that don't have an interpreter

Reply:
so I'll just create methods in Binary one, that don't use interpreter

Reply:
refactored